src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/GroovySandbox.java

```diff
@@ -154,16 +154,12 @@ public Scope enter() {
      * @return the return value of the script
      */
     public Object runScript(@Nonnull GroovyShell shell, @Nonnull String script) {
-        Script s;
-        try (Scope scope = enter()) {
-            s = shell.parse(script);
-        }
         GroovySandbox derived = new GroovySandbox().
             withApprovalContext(context).
             withTaskListener(listener).
-            withWhitelist(new ProxyWhitelist(new ClassLoaderWhitelist(s.getClass().getClassLoader()), whitelist()));
+            withWhitelist(new ProxyWhitelist(new ClassLoaderWhitelist(shell.getClassLoader()), whitelist()));
```

Comment:
A bit awkward, because this only fixes the issue if the caller has modified `shell.loader` reflectively to make it something like `CleanGroovyClassLoader`, but we don't have a good way to enforce that condition. Should we provide a `GroovySandbox.prepareGroovyShell` method as well that sets up `CleanGroovyClassLoader` for other callers?

Reply:
Also if there are any direct callers doing something like SandboxInterceptorTest that aren't using the `CleanGroovyClassLoader` trick, then I think this is a breaking change that will block any types defined in the script because `shell.getClassloader()` will be different than `s.getClass().getClassLoader()`. That said, I don't see any existing callers of `runScript` outside of this plugin.

Reply:
> this only fixes the issue

I.e., the regression. So long as a misuse of this method does not result in a _vulnerability_, I think it is good enough.